source/blog/2017-11-01-async-stack-traces.md

```diff
@@ -0,0 +1,115 @@
+---
+layout: post
+title: Introducing Twitter Futures Capture Points
+published: true
+post_author:
+  display_name: Stefan Lance
+  twitter: stefan_lance
+tags: Util
+---
+
+If you've ever tried to debug asynchronous code written with Twitter `Future`s,
+you may have found that the stack trace isn't always helpful.  The stack trace
+at a given point in the program includes the active stack frames and thus shows
+the chain of function calls that led up to that point.  Since asynchronous
+execution causes different parts of a program to run on different threads, the
+stack trace often gets mangled: it shows unexpected internal library calls and
```

Comment:
i think mangled is the wrong word choice—the stacktraces are always correct. 

instead i think a good way of thinking about it is — the stacks are chopped up and you've lost the context that lead you to the current stack frame.